### PR TITLE
Add apiserver pdb

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Adds PodDisruptionBudget for the Apiserver deployment.
It uses `maxUnavailable=1` as `minAvailable` is not possible when using 1 replica - See https://github.com/kubernetes/kubernetes/issues/66811

The PodDisruptionBudget helps in scenarios with multiple Apiservers (Possible by overriding the replicas via `cluster.spec.componentOverride`)

**Release note**:
```release-note
PodDisruptionBudget is now configured for the API server deployment
```

### CODE REVIEW

File: api/pkg/resources/apiserver/pdb.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// PodDisruptionBudget returns the apiserver PodDisruptionBudget
+func PodDisruptionBudget(data *resources.TemplateData, existing *policyv1beta1.PodDisruptionBudget) (*policyv1beta1.PodDisruptionBudget, error) {
+	var pdb *policyv1beta1.PodDisruptionBudget
+	if existing != nil {
+		pdb = existing
+	} else {
+		pdb = &policyv1beta1.PodDisruptionBudget{}
+	}
+
+	pdb.Name = resources.ApiserverPodDisruptionBudgetName
```
Comment (alvaroaleman) [inline api/pkg/resources/apiserver/pdb.go:20]: This must have a namespace

Comment (mrIncompetent) [reply api/pkg/resources/apiserver/pdb.go:20]: Namespace gets set one level above: https://github.com/kubermatic/kubermatic/blob/add-apiserver-pdb/api/pkg/controller/cluster/resources.go#L376

Comment (alvaroaleman) [reply api/pkg/resources/apiserver/pdb.go:20]: Can we then update the tests to use that func in order to align the fixtures with what we actually deploy?

Comment (mrIncompetent) [reply api/pkg/resources/apiserver/pdb.go:20]: We pass in the namespaced client - so the namespace will never land in the manifest/fixture.
Same behavior with all other manifests/fixtures we have

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**